frontend/packages/dev-console/src/components/import/jar/UploadJar.tsx

```diff
@@ -0,0 +1,228 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { AlertVariant } from '@patternfly/react-core';
+import { useExtensions, Perspective, isPerspective } from '@console/plugin-sdk';
+import {
+  ALL_APPLICATIONS_KEY,
+  getOwnedResources,
+  useActivePerspective,
+  useToast,
+  useBeforeUnloadWatchResource,
+} from '@console/shared/src';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+  WatchK8sResultsObject,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { history, resourcePathFromModel } from '@console/internal/components/utils';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { sanitizeApplicationValue } from '@console/topology/src/utils';
+import { BuildModel, BuildConfigModel } from '@console/internal/models';
+import { Resources, UploadJarFormData } from '../import-types';
+import { healthChecksProbeInitialData } from '../../health-checks/health-checks-probe-utils';
+import UploadJarForm from './UploadJarForm';
+import { BuilderImage } from '../../../utils/imagestream-utils';
+import { createOrUpdateJarFile } from '../upload-jar-submit-utils';
+import { handleRedirect } from '../import-submit-utils';
+import { validationSchema } from '../upload-jar-validation-utils';
+
+export type UploadJarProps = {
+  namespace: string;
+  projects: WatchK8sResultsObject<K8sResourceKind[]>;
+  builderImage: BuilderImage;
+  forApplication?: string;
+};
+const UploadJar: React.FunctionComponent<UploadJarProps> = ({
+  namespace,
+  projects,
+  forApplication,
+  builderImage,
+}) => {
+  const { t } = useTranslation();
+  const [perspective] = useActivePerspective();
+  const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
+  const toastContext = useToast();
+  const beforeUnloadWatchResourceContext = useBeforeUnloadWatchResource();
+  const buildsResource: WatchK8sResource = {
+    kind: BuildModel.kind,
+    namespace,
+    isList: true,
+  };
+  const [builds] = useK8sWatchResource<K8sResourceKind[]>(buildsResource);
+
+  const application = forApplication || '';
+  const activeApplication = application !== ALL_APPLICATIONS_KEY ? application : '';
+  const { name: imageName, recentTag: tag } = builderImage;
+
+  const initialValues: UploadJarFormData = {
```

Comment:
We could extract the common shared fields from this UploadJar and gitImportForm's initialValues and extend that object in respective places to avoid code duplication.

Reply:
updated